core/primitives/src/block_header.rs

```diff
@@ -714,7 +713,7 @@ impl BlockHeader {
 
     /// Common logic for generating BlockHeader for different purposes, including new blocks, from views, and for genesis block
     fn new_impl(
-        this_epoch_protocol_version: ProtocolVersion,
+        _this_epoch_protocol_version: ProtocolVersion,
```

Comment:
(aside) We would like to keep this parameter here for future checks on protocol version. It's a headache to pipe through this parameter.